shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -549,8 +550,20 @@ bool FlutterWindowsEngine::SendPlatformMessage(
       response_handle,
   };
 
-  FlutterEngineResult message_result =
-      embedder_api_.SendPlatformMessage(engine_, &platform_message);
+  FlutterEngineResult message_result;
+  if (task_runner_->RunsTasksOnCurrentThread()) {
+    message_result =
+        embedder_api_.SendPlatformMessage(engine_, &platform_message);
+  } else {
+    fml::AutoResetWaitableEvent latch;
+    task_runner_->PostTask([&]() {
+      message_result =
+          embedder_api_.SendPlatformMessage(engine_, &platform_message);
+      latch.Signal();
+    });
+    latch.Wait();
```

Comment:
If blocking is not desired, the signatures of `FlutterDesktopMessengerSend`/`FlutterDesktopMessengerSendWithReply` would need to be changed in order to accept a callback.
Besides, the message payload and channel name would need to be copied.

Reply:
Having dealt with deadlocks caused by `dispatch_sync`, this does raise my eyebrows a bit. There is some precedent for blocking threads already, for example for resize synchronization, but that's usually as a last resort because there's no other way to accomplish this, rather than a convenience, as it seems to be the case here.

Reply:
I agree and will wait for a decision being made regardings these options: https://github.com/flutter/flutter/issues/134346#issuecomment-1807665020

Reply:
@jnschulze Could you expand why the signatures would need to be changed to remove this blocking?

The engine's `SendPlatformMessage` [fires a task to the UI thread without blocking](https://github.com/flutter/engine/blob/bb4ec2d7eb399f1c878b5342d502deb8214af407/shell/platform/embedder/embedder_platform_message_response.cc#L26-L31). It would seem reasonable to post a task to the platform thread without blocking too.